ionc/ion_double.c

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2009-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at:
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+#include <ionc/ion.h>
+#include <math.h>
+
+BOOL ion_double_is_negative_zero(double value) {
+    return value == 0.0 && signbit(value);
+}
```

Comment:
Let's just add this to the existing ion_float.c/h.

In accordance with my other comments about naming, we'll end up with

```c
BOOL ion_float_is_negative_zero(double value);
BOOL ion_float_32_is_negative_zero(float value);
```

I think not adding a new header will also fix the error on the CI builds. There was probably another place that the new header had to be declared in the CMake configuration...

Reply:
Ah, right, I think I missed a CMake, but will remove the new file.